test/tools/d_do_test.d

```diff
@@ -1550,3 +1557,26 @@ void printTestFailure(string testLogName, string output_file_temp)
     writeln("==============================");
     remove(output_file_temp);
 }
+
+/**
+ * Print symbols in C++ objects
+ *
+ * If linking failed, we print the symbols present in the C++ object file being
+ * linked it. This is so that C++ `runnable` tests are easier to debug,
+ * as the CI machines can have different environment than the users,
+ * and it is generally painful to work with them when trying to support
+ * newer (C++11, C++14, C++17, etc...) features.
+ */
+void printCppSources (in const(char)[][] compiled)
+{
+    version (Posix)
```

Comment:
The windows machines have some ported Posix utilities AFAICT (e.g. bash). Might be worth to try `nm` and silently ignore failures on windows.